From eb9b458d28b654b4d688f27ae3188da15be01f6f Mon Sep 17 00:00:00 2001
From: Davidlohr Bueso <dave@stgolabs.net>
Date: Thu, 1 Dec 2016 07:13:42 -0800
Subject: [PATCH] locking/mutex: Explicitly mark task as running after wakeup
Git-commit: 51587bcf31d070119d37de6103543c807f5ccdb3
Patch-mainline: v4.0-rc1
References: bsc#1012411

By the time we wake up and get the lock after being asleep
in the slowpath, we better be running. As good practice,
be explicit about this and avoid any mischief.

Signed-off-by: Davidlohr Bueso <dbueso@suse.de>
Signed-off-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Cc: "Paul E. McKenney" <paulmck@linux.vnet.ibm.com>
Cc: Linus Torvalds <torvalds@linux-foundation.org>
Link: http://lkml.kernel.org/r/1421717961.4903.11.camel@stgolabs.net
Signed-off-by: Ingo Molnar <mingo@kernel.org>

---
 kernel/mutex.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/kernel/mutex.c b/kernel/mutex.c
index f8841c659dc3..041f6e0313c0 100644
--- a/kernel/mutex.c
+++ b/kernel/mutex.c
@@ -503,6 +503,8 @@ slowpath:
 		preempt_disable();
 		spin_lock_mutex(&lock->wait_lock, flags);
 	}
+	__set_task_state(task, TASK_RUNNING);
+
 	mutex_remove_waiter(lock, &waiter, current_thread_info());
 	/* set it to 0 if there are no waiters left: */
 	if (likely(MUTEX_LIST_EMPTY(&lock->wait_list)))
-- 
2.6.6

